src/cms/models/push_notifications/push_notification_translation.py

```diff
@@ -19,8 +19,8 @@ class PushNotificationTranslation(models.Model):
                      ``push_notification_translations``)
     """
 
-    title = models.CharField(max_length=250)
-    text = models.CharField(max_length=250)
+    title = models.CharField(max_length=250, blank=True)
+    text = models.CharField(max_length=250, blank=True)
```

Comment:
Are you sure that we should allow empty fields here?

Reply:
Well, at least the current Wordpress behavior is this: if a translation for a language does not exist, the source language will be used. And if I do not support blank title & text, then the formset validation will almost always fail.

Reply:
But wouldn't it be better not to save empty translations instead of storing empty objects in the database? Also, we can't use the source language if we "think" that the translation exists (and it's just empty)...
But if there's no way to save only the forms of a formset which were valid, I guess we can live with this downside.

Reply:
The alternative is overriding the validation checks. Not sure if there is any advantage doing it this way. In both options I'd need to independently check if there is a `title` or `text` string available, and if not, use the source language for one or both of them.

Reply:
Should it really be an option to half-translate a PN? Imo, if e.g. the title is translated and the body isn't, I wouldn't send PNs with mixed languages, but raise an error so the sender can fill in both fields. Only if none of them is filled, it makes sense to fallback to the source language...
And then it does make a difference, because it is cleaner to check whether an object exists in the database vs. to check whether an existing object has only empty fields...

Reply:
There are more problems: if the model is invalid with a missing title, then handling the formset becomes complicated. When loading the formset for editing, it is easy to load existing objects for a model into the formset. However, we would then have to compute which languages are missing and create an initial dict to fill the vacant translations. While this is possible, I'd argue its much easier to understand (and compute) to simply store empty translations. And because we actually allow empty translations, this is not that wrong at all. Isn't it?